sky/packages/sky/lib/src/gestures/events.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' as ui;
+
+/// Base class for input events.
+class InputEvent {
+
+  const InputEvent({ this.type, this.timeStamp: 0.0 });
+
+  factory InputEvent.fromUiEvent(ui.Event event) {
+    if (event is ui.PointerEvent)
+      return new PointerInputEvent.fromUiEvent(event);
+
+    // Default event
+    return new InputEvent(
+      type: event.type,
+      timeStamp: event.timeStamp
+    );
+  }
+
+  final String type;
+  final double timeStamp;
+
+}
+
+/// Input event representing a touch or button.
+class PointerInputEvent extends InputEvent {
+
+  const PointerInputEvent({
+    String type,
```

Comment:
"type" is temporary, it's just to make it trivial to convert ui.Event to this new data structure.